salt/utils/dictupdate.py

```diff
@@ -50,7 +54,7 @@ def update(dest, upd, recursive_update=True, merge_lists=False):
             elif isinstance(dest_subkey, list) \
                      and isinstance(val, list):
                 if merge_lists:
-                    dest[key] = dest.get(key, []) + val
+                    dest[key] = sorted((set(dest_subkey + val)))
```

Comment:
Ok, I'm not hung up on the sort detail. I'll pull that out and massage the docstring a bit to say which list is merged into which list, and which list the dupes will be removed from. (That's the decision I was avoiding by sorting.)